topology/m4/mixercontrol.m4

```diff
@@ -28,6 +28,47 @@ define(`CONTROLMIXER_OPS',
 `		put STR($4)'
 `	}')
 
+dnl KCONTROL_2CHANNELS(reg)
+define(`KCONTROL_2CHANNELS',
+`KCONTROL_CHANNEL(FL, $1, 0),'
+`KCONTROL_CHANNEL(FR, $1, 1)')
+
+dnl KCONTROL_4CHANNELS(reg)
+define(`KCONTROL_4CHANNELS',
+`KCONTROL_CHANNEL(FL, $1, 0),'
+`KCONTROL_CHANNEL(FR, $1, 1),'
+`KCONTROL_CHANNEL(RL, $1, 2),'
+`KCONTROL_CHANNEL(RR, $1, 3)')
+
+dnl KCONTROL_6CHANNELS(reg)
+define(`KCONTROL_6CHANNELS',
+`KCONTROL_CHANNEL(FL, $1, 0),'
+`KCONTROL_CHANNEL(FR, $1, 1),'
+`KCONTROL_CHANNEL(RL, $1, 2),'
+`KCONTROL_CHANNEL(RR, $1, 3),'
+`KCONTROL_CHANNEL(FC, $1, 4),'
+`KCONTROL_CHANNEL(LFE, $1, 5)')
+
+dnl KCONTROL_8CHANNELS(reg)
+define(`KCONTROL_8CHANNELS',
+`KCONTROL_CHANNEL(FL, $1, 0),'
+`KCONTROL_CHANNEL(FR, $1, 1),'
+`KCONTROL_CHANNEL(RL, $1, 2),'
+`KCONTROL_CHANNEL(RR, $1, 3),'
+`KCONTROL_CHANNEL(FC, $1, 4),'
+`KCONTROL_CHANNEL(LFE, $1, 5),'
+`KCONTROL_CHANNEL(SL, $1, 6),'
+`KCONTROL_CHANNEL(SR, $1, 7)')
+
+dnl KCONTROL_CHANNEL_ARRAY(size, reg)
+define(`KCONTROL_CHANNEL_ARRAY',
+`ifelse(
+        $1, `2', `KCONTROL_2CHANNELS($2)',
```

Comment:
what about odd number of channels. it's perfectly doable in TDM mode...

Reply:
Thanks, I will refine it